fix server build errors

```
% npm run build

> server@1.0.0 build
> tsc

src/models/chatThread.ts:24:12 - error TS2339: Property 'initModel' does not exist on type 'typeof ChatThread'.

24 ChatThread.initModel(
              ~~~~~~~~~

src/models/user.ts:20:6 - error TS2339: Property 'initModel' does not exist on type 'typeof User'.

20 User.initModel(
        ~~~~~~~~~

src/routes/auth.ts:2:20 - error TS7016: Could not find a declaration file for module 'bcrypt'. '/Users/hideya/Desktop/RA/ws/test-chatly-ai-gw/server/node_modules/bcrypt/bcrypt.js' implicitly has an 'any' type.
  Try `npm i --save-dev @types/bcrypt` if it exists or add a new declaration (.d.ts) file containing `declare module 'bcrypt';`

2 import bcrypt from 'bcrypt';
                     ~~~~~~~~

src/routes/auth.ts:3:17 - error TS7016: Could not find a declaration file for module 'jsonwebtoken'. '/Users/hideya/Desktop/RA/ws/test-chatly-ai-gw/server/node_modules/jsonwebtoken/index.js' implicitly has an 'any' type.
  Try `npm i --save-dev @types/jsonwebtoken` if it exists or add a new declaration (.d.ts) file containing `declare module 'jsonwebtoken';`

3 import jwt from 'jsonwebtoken';
                  ~~~~~~~~~~~~~~

src/routes/auth.ts:20:29 - error TS2339: Property 'createUser' does not exist on type 'typeof User'.

20     const user = await User.createUser({ username, password: hashedPassword });
                               ~~~~~~~~~~

src/routes/auth.ts:35:29 - error TS2339: Property 'findUser' does not exist on type 'typeof User'.

35     const user = await User.findUser({ where: { username } });
                               ~~~~~~~~

src/routes/chat.ts:17:41 - error TS2339: Property 'createThread' does not exist on type 'typeof ChatThread'.

17     const chatThread = await ChatThread.createThread({ userId, messages: [{ sender: 'user', text: message }] });
                                           ~~~~~~~~~~~~

src/routes/chat.ts:36:41 - error TS2339: Property 'findThreadByPk' does not exist on type 'typeof ChatThread'.

36     const chatThread = await ChatThread.findThreadByPk(threadId);
                                           ~~~~~~~~~~~~~~

src/routes/chat.ts:57:42 - error TS2339: Property 'findAllThreads' does not exist on type 'typeof ChatThread'.

57     const chatThreads = await ChatThread.findAllThreads({ where: { userId }, order: [['createdAt', 'DESC']] });
                                            ~~~~~~~~~~~~~~

src/routes/chat.ts:68:41 - error TS2339: Property 'findThreadByPk' does not exist on type 'typeof ChatThread'.

68     const chatThread = await ChatThread.findThreadByPk(threadId);
                                           ~~~~~~~~~~~~~~

src/services/openai.ts:18:14 - error TS2532: Object is possibly 'undefined'.

18       return response.data.choices[0].text.trim();
                ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~


Found 11 errors in 5 files.

Errors  Files
     1  src/models/chatThread.ts:24
     1  src/models/user.ts:20
     4  src/routes/auth.ts:2
     4  src/routes/chat.ts:17
     1  src/services/openai.ts:18
```

--- a/server/package.json
+++ b/server/package.json
@@ -21,6 +21,8 @@
     "typescript": "^4.3.5",
     "ts-node": "^10.0.0",
     "@types/express": "^4.17.11",
-    "@types/node": "^14.14.37"
+    "@types/node": "^14.14.37",
+    "@types/bcrypt": "^5.0.0",
+    "@types/jsonwebtoken": "^8.5.0"
   }
 }

--- a/server/src/models/chatThread.ts
+++ b/server/src/models/chatThread.ts
@@ -19,38 +19,52 @@ class ChatThread extends Model<ChatThreadAttributes, ChatThreadCreationAttribute
   public messages!: { sender: string; text: string }[];
   public createdAt!: Date;
   public updatedAt!: Date;
-}
 
-ChatThread.initModel(
-  {
-    id: {
-      type: DataTypes.INTEGER,
-      autoIncrement: true,
-      primaryKey: true,
-    },
-    userId: {
-      type: DataTypes.INTEGER,
-      allowNull: false,
-    },
-    messages: {
-      type: DataTypes.JSON,
-      allowNull: false,
-    },
-    createdAt: {
-      type: DataTypes.DATE,
-      allowNull: false,
-      defaultValue: DataTypes.NOW,
-    },
-    updatedAt: {
-      type: DataTypes.DATE,
-      allowNull: false,
-      defaultValue: DataTypes.NOW,
-    },
-  },
-  {
-    sequelize,
-    modelName: 'ChatThread',
+  public static initModel(sequelize: Sequelize) {
+    ChatThread.init(
+      {
+        id: {
+          type: DataTypes.INTEGER,
+          autoIncrement: true,
+          primaryKey: true,
+        },
+        userId: {
+          type: DataTypes.INTEGER,
+          allowNull: false,
+        },
+        messages: {
+          type: DataTypes.JSON,
+          allowNull: false,
+        },
+        createdAt: {
+          type: DataTypes.DATE,
+          allowNull: false,
+          defaultValue: DataTypes.NOW,
+        },
+        updatedAt: {
+          type: DataTypes.DATE,
+          allowNull: false,
+          defaultValue: DataTypes.NOW,
+        },
+      },
+      {
+        sequelize,
+        modelName: 'ChatThread',
+      }
+    );
   }
-);
+
+  public static async createThread(data: ChatThreadCreationAttributes): Promise<ChatThread> {
+    return await ChatThread.create(data);
+  }
+
+  public static async findThreadByPk(id: number): Promise<ChatThread | null> {
+    return await ChatThread.findByPk(id);
+  }
+
+  public static async findAllThreads(options: any): Promise<ChatThread[]> {
+    return await ChatThread.findAll(options);
+  }
+}
 
 export { ChatThread };

--- a/server/src/models/user.ts
+++ b/server/src/models/user.ts
@@ -15,29 +15,39 @@ class User extends Model<UserAttributes, UserCreationAttributes> implements User
   public id!: number;
   public username!: string;
   public password!: string;
-}
 
-User.initModel(
-  {
-    id: {
-      type: DataTypes.INTEGER,
-      autoIncrement: true,
-      primaryKey: true,
-    },
-    username: {
-      type: DataTypes.STRING,
-      allowNull: false,
-      unique: true,
-    },
-    password: {
-      type: DataTypes.STRING,
-      allowNull: false,
-    },
-  },
-  {
-    sequelize,
-    modelName: 'User',
+  public static initModel(sequelize: Sequelize) {
+    User.init(
+      {
+        id: {
+          type: DataTypes.INTEGER,
+          autoIncrement: true,
+          primaryKey: true,
+        },
+        username: {
+          type: DataTypes.STRING,
+          allowNull: false,
+          unique: true,
+        },
+        password: {
+          type: DataTypes.STRING,
+          allowNull: false,
+        },
+      },
+      {
+        sequelize,
+        modelName: 'User',
+      }
+    );
   }
-);
+
+  public static async createUser(data: UserCreationAttributes): Promise<User> {
+    return await User.create(data);
+  }
+
+  public static async findUser(options: any): Promise<User | null> {
+    return await User.findOne(options);
+  }
+}
 
 export { User };

--- a/server/src/services/openai.ts
+++ b/server/src/services/openai.ts
@@ -15,7 +15,7 @@ export const getOpenAIResponse = async (message: string): Promise<string> => {
     });
 
     if (response.data.choices && response.data.choices.length > 0) {
-      return response.data.choices[0].text.trim();
+      return response.data.choices[0].text?.trim() ?? '';
     } else {
       throw new Error('No choices found in OpenAI response');
     }

